src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -146,6 +146,10 @@ def implementation_version(cls):
   def register_options(cls, register):
     super(JUnitRun, cls).register_options(register)
 
+    register('--fast', type=bool, default=True, fingerprint=True,
+             help='Run all tests in a single junit invocation. If turned off, each test target '
+                  'will run in its own junit invocation, which will be slower, but isolates '
+                  'tests from process-wide state created by tests in other targets.')
     register('--batch-size', advanced=True, type=int, default=cls._BATCH_ALL, fingerprint=True,
              help='Run at most this many tests in a single test process.')
```

Comment:
Worth documenting the relation of this flag to `--fast`? ...and maybe deprecating it in its current form?

Reply:
I'd like to defer all the option work to #5073 if that works for you. Right now you can `--no-fast` and still want to use `--batch-size=1` if, say, you had 1 target with tons of test classes, so seems fine to leave both in this commit since the combos are handled.